File: java-compat/src/Ice/src/main/java/IceInternal/WSEndpointFactory.java
// **********************************************************************
//
// Copyright (c) 2003-present ZeroC, Inc. All rights reserved.
//
// **********************************************************************

package IceInternal;

final public class WSEndpointFactory extends EndpointFactoryWithUnderlying
{
    public WSEndpointFactory(ProtocolInstance instance, short type)
    {
        super(instance, type);
    }

    @Override
    public EndpointFactory cloneWithUnderlying(ProtocolInstance instance, short underlying)
    {
        return new WSEndpointFactory(instance, underlying);
    }

    @Override
    public EndpointI createWithUnderlying(EndpointI underlying, java.util.ArrayList<String> args, boolean oaEndpoint)
    {
        return new WSEndpoint(_instance, underlying, args);
    }

    @Override
    public EndpointI readWithUnderlying(EndpointI underlying, Ice.InputStream s)
    {
        return new WSEndpoint(_instance, underlying, s);
    }
}

File: js/packages/ice/src/Ice/WSEndpointFactory.js
// Copyright (c) ZeroC, Inc.

import { WSEndpoint } from "./WSEndpoint.js";

export class WSEndpointFactory {
    constructor(instance, delegate) {
        this._instance = instance;
        this._delegate = delegate;
    }

    type() {
        return this._instance.type();
    }

    protocol() {
        return this._instance.protocol();
    }

    create(args, oaEndpoint) {
        const e = new WSEndpoint(this._instance, this._delegate.create(args, oaEndpoint));
        e.initWithOptions(args);
        return e;
    }

    read(s) {
        const e = new WSEndpoint(this._instance, this._delegate.read(s));
        e.initWithStream(s);
        return e;
    }
}
